backend/api/messaging/index.js

```diff
@@ -42,9 +42,14 @@ router.post(
     }
 
     const { sid } = clients.getReceiverByChannel(channel, sender);
-    socketEmit('chat-message', sid, dataToPublish);
+    const received = socketEmit('chat-message', sid, dataToPublish);
+    if (received) {
+      const senderID = clients.getSenderByChannel(channel, sender);
+      socketEmit('delivered', senderID.sid, {});
+      return res.send({ message: 'message sent' });
+    }
 
-    return res.send({ message: 'message sent' });
+    return res.send(400);
```

Comment:
This doesn't ensure whether the message is delivered to the client device. We need to handle when the client from the receiving end actually received the message.

Reply:
@muke1908 
https://stackoverflow.com/questions/11382495/how-to-be-sure-that-message-via-socket-io-has-been-received-to-the-client

this kind of callback can be used in `socketEmit()` to ensure delivery

Reply:
@nafees87n , you can get data back from callback 

```
socket.on('someEvent', function (data, callback) {
    callback({ok:true});
});

client.emit("someEvent", someData, function (data) {
  // data = `{ok:true}`
});
```